dynamo/preprocessing/pca.py

```diff
@@ -358,3 +358,34 @@ def top_pca_genes(
     main_info_insert_adata_var(adata_store_key, indent_level=2)
     adata.var[adata_store_key] = genes
     return adata
+
+
+def pca_transform(data: Union[np.ndarray, csr_matrix], PCs: np.ndarray) -> np.ndarray:
+    """Transform the data with given principal components.
+
+    Args:
+        data: raw data to transform.
+        PCs: the principal components.
+
+    Returns:
+        The transformed data.
+    """
+    arr = data.toarray().copy() if issparse(data) else data.copy()
+    mean = arr.mean(0)
+    centered_data = arr - mean
+    return np.dot(centered_data, PCs)
+
+
+def pca_inverse_transform(data: Union[np.ndarray, csr_matrix], PCs: np.ndarray, mean: np.ndarray) -> np.ndarray:
+    """Inverse transform the data with given principal components.
+
+    Args:
+        data: raw data to transform.
+        PCs: the principal components.
+        mean: the mean used to fit the PCA.
+
+    Returns:
+        The inverse transformed data.
+    """
+    arr = data.toarray().copy() if issparse(data) else data.copy()
+    return np.dot(arr, PCs.T) + mean
```

Comment:
duplicated code with predictions/utils. uniform them. and just use one 
```

# trajectory related
def pca_to_expr(X, PCs, mean=0, func=None):
    # reverse project from PCA back to raw expression space
    if PCs.shape[1] == X.shape[1]:
        exprs = X @ PCs.T + mean
        if func is not None:
            exprs = func(exprs)
    else:
        raise Exception("PCs dim 1 (%d) does not match X dim 1 (%d)." % (PCs.shape[1], X.shape[1]))
    return exprs


def expr_to_pca(expr, PCs, mean=0, func=None):
    # project from raw expression space to PCA
    if PCs.shape[0] == expr.shape[1]:
        X = (expr - mean) @ PCs
        if func is not None:
            X = func(X)
    else:
        raise Exception("PCs dim 1 (%d) does not match X dim 1 (%d)." % (PCs.shape[0], expr.shape[1]))
    return X

```

Reply:
Done. Now use `expr_to_pca` and `pca_to_expr`.